### ISSUE TITLE
Tiny mouse pointer under wayland

### ISSUE THREAD

**Issue by DavidGasku** [godotengine/godot#96459]
### Tested versions

Godot v4.3.stable

### System information

Godot v4.3.stable - Manjaro Linux #1 SMP PREEMPT_DYNAMIC Mon Aug 19 09:51:26 UTC 2024 - Wayland - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 (nvidia; 550.107.02) - 12th Gen Intel(R) Core(TM) i5-12400F (12 Threads)

### Issue description

[Screencast_20240902_080033.webm](https://github.com/user-attachments/assets/006f0853-1956-44bc-b082-e6356b43b242)

As seen in the video, the mouse is tiny compared with the OS one (the size changes when hovering the system titlebar)

### Steps to reproduce

Just start godot with wayland support (--display-driver wayland) under linux

### Minimal reproduction project (MRP)

Not applicable

**Comment by hunterkepley** [comment]
Could you set the env var `XCURSOR_SIZE` to `35` and see if it makes it larger, or if there is an error. It seems there's a default size of 24 if there's an error, but I want to see if this can be resolved by adding `XCURSOR_SIZE` to your preferred size in your bash.rc. If not, might be good to update the default size from `24` to something larger.

**Comment by hunterkepley** [comment]
Source for this env var: https://github.com/godotengine/godot/blob/61598c5c88d95b96811d386cb20d714c35f4c6d7/platform/linuxbsd/wayland/wayland_thread.cpp#L3753

**Comment by DavidGasku** [comment]
Doesn't seem to make any difference the cursor is always the same size.
Terminal shows no errors (just discovered a segmentation fault on exit, but probably was there before and didn't notice it and is unrelated)

I've tested several scaling values (both in godot and in kde) and the cursor size is actually smaller with higher kde fractional values. First image is 150%, second one is 100%.

![Screenshot_20240902_124809](https://github.com/user-attachments/assets/e87dcf45-2b8b-4450-ab53-771809fc4024)
![Screenshot_20240902_124728](https://github.com/user-attachments/assets/eaad5b92-7e53-4c78-bd98-8d56bfc2f942)

In case is needed, I´m using a single 4K monitor.

**Comment by hunterkepley** [comment]
im gonna upgrade to the newest fedora so i can test out with wayland today, want to take a crack at it. thanks for trying that out

**Comment by hunterkepley** [comment]
@DavidGasku Did some poking around on the internet, this is a common problem with older versions of wayland/specific versions.

https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/5481
https://www.reddit.com/r/kde/comments/133dkad/gtk_mouse_scaling_on_wayland/
https://invent.kde.org/plasma/breeze/-/merge_requests/380
https://github.com/kovidgoyal/kitty/issues/2647
https://github.com/swaywm/sway/issues/4610
https://www.reddit.com/r/swaywm/comments/n2ha58/tiny_cursor_on_chromium_using_ozone_platform/

It seems that GNOME has a merged fix (see the top link in this comment)- but it is platform dependent as well as Qt/GTK dependent.

Not sure if this is something related to Godot, to be honest, and fixing it would be changing your wayland settings through godot, which I believe would be the wrong solution

Could you provide more info about your setup/system? I see you are on manjaro, would this happen to be using Breeze, Gnome, or another DE?

tl;dr I think this issue is not a godot problem, but a wayland problem

**Comment by hunterkepley** [comment]
Cannot reproduce with Fedora 39 - KDE using Wayland (all up-to-date besides fedora version)

**Comment by DavidGasku** [comment]
I'm using KDE. Anyway, feel free to close it if you can't reproduce it. I'll wait on X11 until multiwindow is implemented.

**Comment by Axilotl17** [comment]
I have the same problem. Additionally, the cursor changes themes between different windows.

**Comment by otonashixav** [comment]
Can reproduce on openSUSE Tumbleweed 20250113 running KDE Plasma 6.2.5 with both v4.3 stable and commit 4a4f9339e. My experience is identically flawed except additionally the cursor obviously is using the wrong theme as it's black instead of white.

**Comment by snoopdouglas** [comment]
Managed to miss this when searching for issues before opening #103170. This issue persists on 4.4-beta1.

In my case, setting `XCURSOR_THEME` did the trick - in my case, as I'm using GNOME with the default Adwaita theme, this was:

```sh
XCURSOR_THEME=Adwaita
```

Whether or not the cursor is 'tiny' without setting this environment variable seems to depend on the distro and DE you're using. On Ubuntu 24.04, you just get the light themed cursor rather than the tiny one.

The cursor you're seeing here is actually the default 'HighContrast' X cursor theme, from what I can tell - which only has one (tiny) size.

**Comment by TuTiuTe** [comment]
@snoopdouglas 's solution is a fine workaround, but it's bothersome to make it persistent. You either set a wayland environment variable (there is no standard on how to do that, see this [Archwiki post](https://wiki.archlinux.org/title/Environment_variables#Graphical_environment) for that) or you edit godot's desktop file to include the environment variable
Either way, when you'll want to change cursors, instead of changing one setting in your DE you'll have to remember to change this variable too
Or even more variables if you have the problem on other apps and chose to edit their desktop files (blender has the same issue for me)

**Comment by snoopdouglas** [comment]
@TuTiuTe - you can add persistent system-wide environment variables to `/etc/environment` (though this is a nuclear option).

**Comment by TuTiuTe** [comment]
I just tested godot 4.5-beta3 and I am no longer getting this issue
I was not getting the "tiny mouse pointer" but the default mouse pointer, but they seem to be the same thing
If anyone someone else can confirm it's also fixed for them, we can maybe mark this issue as solved for 4.5 ~and find what fixed it :p 
Seems to be coming from a wayland fix batch since the title bar now follows the system theme and some warning messages about some stuff not working on wayland were removed~

EDIT:
This is probably the PR that fixes this issue #106251
And also this issue is probably a duplicate of #106229
